cmd/promtool/main.go

```diff
@@ -353,8 +353,9 @@ func checkConfig(agentMode bool, filename string) ([]string, error) {
 	}
 
 	for _, scfg := range cfg.ScrapeConfigs {
-		if err := checkFileExists(scfg.HTTPClientConfig.BearerTokenFile); err != nil {
-			return nil, errors.Wrapf(err, "error checking bearer token file %q", scfg.HTTPClientConfig.BearerTokenFile)
+		authzSet := scfg.HTTPClientConfig.Authorization != nil
+		if err := checkFileExists(scfg.HTTPClientConfig.Authorization.CredentialsFile); authzSet && err != nil {
```

Comment:
The compiler seems to be smart enough to not run `err := ...` if `authzSet` is already false. This is quite neat since it avoids using a nested `if` statement.